prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -221,52 +221,81 @@ public APIResult schedule(@PathParam("job-name") String jobName,
     @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
     @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
     public APIResult suspend(@PathParam("job-name") String jobName,
-                             @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+                             @Context HttpServletRequest request,
+                             @DefaultValue("") @QueryParam("doAs") String doAsUser,
+                             @QueryParam("colo") final String coloExpr) {
         checkIfExtensionServiceIsEnabled();
-        try {
-            List<Entity> entities = getEntityList("", "", "", TAG_PREFIX_EXTENSION_JOB + jobName, "", doAsUser);
-            if (entities.isEmpty()) {
-                // return failure if the extension job doesn't exist
-                return new APIResult(APIResult.Status.FAILED, "Extension job " + jobName + " doesn't exist.");
-            }
+        ExtensionMetaStore metaStore = ExtensionStore.getMetaStore();
+        ExtensionJobsBean extensionJobsBean = metaStore.getExtensionJobDetails(jobName);
+        if (extensionJobsBean == null) {
+            // return failure if the extension job doesn't exist
+            LOG.error("Extension Job not found:" + jobName);
+            throw FalconWebException.newAPIException("ExtensionJob not found:" + jobName,
+                    Response.Status.NOT_FOUND);
+        }
 
-            for (Entity entity : entities) {
-                if (entity.getEntityType().isSchedulable()) {
-                    if (getWorkflowEngine(entity).isActive(entity)) {
-                        getWorkflowEngine(entity).suspend(entity);
-                    }
-                }
-            }
-        } catch (FalconException | IOException e) {
-            LOG.error("Error when scheduling extension job: " + jobName + ": ", e);
+        try {
+            suspendEntities(extensionJobsBean, coloExpr, request);
+        } catch (FalconException e) {
+            LOG.error("Error while suspending entities of the extension: " + jobName + ": ", e);
             throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
         }
         return new APIResult(APIResult.Status.SUCCEEDED, "Extension job " + jobName + " suspended successfully");
     }
 
+    private void suspendEntities(ExtensionJobsBean extensionJobsBean, String coloExpr, final HttpServletRequest request)
+        throws FalconException {
+        List<String> processes = extensionJobsBean.getProcesses();
+        List<String> feeds = extensionJobsBean.getFeeds();
+        final HttpServletRequest bufferedRequest = new BufferedRequest(request);
+        suspendEntities(coloExpr, feeds, EntityType.FEED.name(), bufferedRequest);
```

Comment:
Don't see a need for existence of 2 suspendEntities method and invoke it twice, if we followed the same code logic used in schedule.